opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -663,6 +679,10 @@ def tracer_source() -> TracerSource:
         except TypeError:
```

Comment:
To catch exceptions when instantiating an abstract base class

Reply:
Thoughts on catching any exception here? the reason I added this was because I ran into behavior where I wasn't overriding the default tracer, and found it's because I had a bug in my unit test (caused by importllib.reload recreating all instances).

I feel like I'd want more feedback, but maybe that's already handled by instantiating a tracer and raising an exception there.